src/crimson/osd/recovery_backend.h

```diff
@@ -126,6 +126,7 @@ class RecoveryBackend {
 
     crimson::osd::ObjectContextRef obc;
     PullInfo pi;
```

Comment:
what about making `pi` an `optional<PullInfo>` or `unique_ptr<PullInfo>` if it's not always valid?

Reply:
Oh, yes!